doc/website/advanced/how-optional-and-error-values-are-returned-in-iceoryx.md

```diff
@@ -0,0 +1,104 @@
+# How optional and error values are returned in iceoryx
+
+Many parts of the iceoryx C++ API follow a functional programming approach and allow the user to specify functions 
+which handle the possible cases, e.g. what should happen when data is received.
+
+This is very flexible but requires using the monadic types ``cxx::expected`` and ``cxx::optional``, which we
+introduce in the following sections.
+
+### Optional
+
+The type ``iox::cxx::optional<T>`` is used to indicate that there may or may not be a value of a specific type ``T``
+available. This is essentially the maybe [monad](https://en.wikipedia.org/wiki/Monad_(functional_programming)) in 
```

Comment:
The maybe [monad](https://en.wikipedia.org/wiki/Monad_(functional_programming)) could be set in quotes... maybe